cmd/olfullnode/save_state.go

```diff
@@ -336,6 +340,31 @@ func DumpFeesToFile(st *fees.Store, writer io.Writer, fn func(writer io.Writer,
 	return
 }
 
+func DumpWitnessToFile(ws *identity.WitnessStore, writer io.Writer, fn func(writer io.Writer, obj interface{}) bool) {
```

Comment:
I would make a ethereum utility file, and add all these ethereum specific changes in that file. In this place, I would iterate all the chain specific utility files, and call the same DumpWitnessToFile method. That way when we add a new chain of witnesses, we dont need to change at multiple places

Reply:
I'd rather do that when we have bitcoin witness, since we are not sure the witness object will change or not for the btc.

Reply:
and this is the only place that dump for witness is call and used, if we create another file, we actually need to change in multiple place, and this function is update only when new chain is added for witness.